packages/firestore/src/platform_browser/browser_connectivity_monitor.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * @license
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { debug } from '../util/log';
+import {
+  ConnectivityMonitor,
+  ConnectivityMonitorCallback,
+  NetworkStatus
+} from './../remote/connectivity_monitor';
+
+const LOG_TAG = 'ConnectivityMonitor';
+
+/**
+ * Browser implementation of ConnectivityMonitor.
+ */
+export class BrowserConnectivityMonitor implements ConnectivityMonitor {
+  private readonly networkAvailableListener = () => this.onNetworkAvailable();
+  private readonly networkUnavailableListener = () =>
+    this.onNetworkUnavailable();
+  private callbacks: ConnectivityMonitorCallback[] = [];
+
+  constructor() {
+    this.configureNetworkMonitoring();
+  }
+
+  addCallback(callback: (status: NetworkStatus) => void): void {
+    this.callbacks.push(callback);
+  }
+
+  shutdown(): void {
+    window.removeEventListener('online', this.networkAvailableListener);
+    window.removeEventListener('offline', this.networkUnavailableListener);
+  }
+
+  private configureNetworkMonitoring(): void {
+    window.addEventListener('online', this.networkAvailableListener);
```

Comment:
@thebrianchen I believe we need to fix this still as per our earlier discussion.

Ideally, you would change BrowserWindow to only return BrowserConnectivityMonitor if `window` is defined. The smaller fix would be to add a `typeof window !== undefined` check here. As merged, I believe this change will cause issues on Electron and other browser-like environments.

Reply:
This is probably causing issue #1824, but I'm not sure though.